### PR TITLE
Minor additions for support SAN/dnsName for restricted trust

### PR DESCRIPTION
**Pull request by jakelandis**
A follow up to https://github.com/elastic/elasticsearch/pull/91946 with the
minor requested changes. 

Changes included here are: 
* reuse of variables
* additional unit test
* convert to use enumeration instead of set of strings

### CODE REVIEW

**Comment by justincr-elastic** [comment]
Why use v8.6.0 instead of v8.6.1? Should versions be v8.7.0, v8.6.1, and v7.17.8?

**Comment by jakelandis** [comment]
If you have already reviewed, please note I have added conversion to an enumeration as part of this change. See [`e7fa550` (#91983)](https://github.com/elastic/elasticsearch/pull/91983/commits/e7fa550330b1132559cfc9b87a97df3e7d77b217)

**Comment by jakelandis** [comment]
Also, I do not plan to backport the enumeration change to 7.17 due to timing and complexity (the 7.17 code base is quite different w.r.t. how the settings are defined)

**Review by tvernum — APPROVED** [review]

File: libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslConfigurationKeys.java [hunk 1]
```diff
@@ -71,6 +71,12 @@ public class SslConfigurationKeys {
      */
     public static final String TRUSTSTORE_ALGORITHM = "truststore.algorithm";
 
+    /**
+     * The fields from the X509 certificate used for restricted trust. Internationally omitted from the list of setting returned by methods
+     * in this class. This is not a general purpose ssl configuration.
+     */
+    public static final String TRUST_RESTRICTIONS_X509_FIELDS = "trust_restrictions.x509_fields";
```
[the remark this answers is not in the document]
Comment (jakelandis) [reply libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslConfigurationKeys.java:78]: SslConfigurationLoader needs to reference this field, which is also in `libs/ssl-config` . I agree this is weird, but then again I feel that I have been struggling with trying to honor the expectations of the settings vs. expectations of the ssl settings. I think it's a larger conversation than this PR, but IMO introducing this setting - the right way -  is much harder than it should be.

**Comment by jakelandis** [comment]
@elasticsearchmachine update branch

**Pull request merged**